common/list.h

```diff
@@ -94,5 +94,7 @@ int
 list_append_list_strdup(struct list *self, struct list *dest, int start_index);
 void
 list_dump_items(struct list *self);
+struct list *
+split_string_into_list(const char *str, char character);
 
```

Comment:
Can do with a header comment saying NULL is returned on allocation error, like some of the other functions here (if you implement the above).

Reply:
Updated!